core/trino-main/src/test/java/io/trino/execution/TestCreateTableTask.java

```diff
@@ -349,6 +357,38 @@ public void testUnsupportedCreateTableWithField()
                 .hasMessage("Column name 'a.b' must not be qualified");
     }
 
+    @Test
+    public void testCreateTableWithCoercedType()
+    {
+        CreateTable statement = new CreateTable(QualifiedName.of("test_table"),
```

Comment:
Can we move to something like this 

        CreateTable statement = new CreateTable(QualifiedName.of("test_table"),
                ImmutableList.of(
                        new ColumnDefinition(
                                QualifiedName.of("a"), 
                                toSqlType(TIMESTAMP_NANOS), 
                                true, 
                                emptyList(), 
                                Optional.empty())),
                IGNORE,
                ImmutableList.of(),
                Optional.empty());